hal/src/thumbv6m/adc.rs

```diff
@@ -59,18 +68,37 @@ impl Adc<ADC> {
         while self.adc.status.read().syncbusy().bit_is_set() {}
     }
 
+    /// Set the gain factor
     pub fn gain(&mut self, gain: adc::inputctrl::GAIN_A) {
         self.adc.inputctrl.modify(|_, w| w.gain().variant(gain));
         while self.adc.status.read().syncbusy().bit_is_set() {}
     }
 
+    /// Set the voltage reference source
     pub fn reference(&mut self, reference: adc::refctrl::REFSEL_A) {
         self.adc
             .refctrl
             .modify(|_, w| w.refsel().variant(reference));
         while self.adc.status.read().syncbusy().bit_is_set() {}
     }
 
+    /// Set the prescaler for adjusting the clock relative to the system clock
+    pub fn prescaler(&mut self, prescaler: adc::ctrlb::PRESCALER_A) {
```

Comment:
I would suggest adding a type alias for the `PRESCALAR_A` type, so that users can import it from this module. Otherwise they'll have to import it from the PAC directly. You can also give it a better name and one that is more consistent with Rust naming conventions.

Maybe you could do the same for the other types in this module as well.